substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/JfrGCHeapSummaryEventSupport.java

```diff
@@ -0,0 +1,121 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package com.oracle.svm.core.genscavenge;
+
+import org.graalvm.nativeimage.ImageSingletons;
+import org.graalvm.nativeimage.StackValue;
+import org.graalvm.word.UnsignedWord;
+
+import com.oracle.svm.core.SubstrateOptions;
+import com.oracle.svm.core.Uninterruptible;
+import com.oracle.svm.core.feature.AutomaticallyRegisteredFeature;
+import com.oracle.svm.core.feature.InternalFeature;
+import com.oracle.svm.core.jfr.HasJfrSupport;
+import com.oracle.svm.core.jfr.JfrEvent;
+import com.oracle.svm.core.jfr.JfrGCWhen;
+import com.oracle.svm.core.jfr.JfrGCWhens;
+import com.oracle.svm.core.jfr.JfrNativeEventWriter;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterData;
+import com.oracle.svm.core.jfr.JfrNativeEventWriterDataAccess;
+
+class JfrGCHeapSummaryEventSupport {
+
+    private final JfrGCWhen before;
+    private final JfrGCWhen after;
+
+    JfrGCHeapSummaryEventSupport(JfrGCWhen before, JfrGCWhen after) {
+        this.before = before;
+        this.after = after;
+    }
+
+    public void emitGCHeapSummaryEventBeforeGC(UnsignedWord gcEpoch, long start, long heapUsed) {
+
+        emitGCHeapSummaryEvent(gcEpoch, start, heapUsed, before);
+
+    }
+
+    public void emitGCHeapSummaryEventAfterGC(UnsignedWord gcEpoch, long start, long heapUsed) {
+
+        emitGCHeapSummaryEvent(gcEpoch, start, heapUsed, after);
+
+    }
+
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    public void emitGCHeapSummaryEvent(UnsignedWord gcEpoch, long start, long heapUsed, JfrGCWhen gcWhen) {
+
+        if (JfrEvent.GCHeapSummary.shouldEmit()) {
+
+            JfrNativeEventWriterData data = StackValue.get(JfrNativeEventWriterData.class);
+            JfrNativeEventWriterDataAccess.initializeThreadLocalNativeBuffer(data);
+
+            JfrNativeEventWriter.beginSmallEvent(data, JfrEvent.GCHeapSummary);
+
+            JfrNativeEventWriter.putLong(data, start); // @Label("Start Time") @Timestamp("TICKS")
+                                                      // long startTime;
+
+            JfrNativeEventWriter.putLong(data, gcEpoch.rawValue()); // @Label("GC Identifier") int
+                                                                   // gcId;
+            JfrNativeEventWriter.putLong(data, gcWhen.getId()); // @Label("When") String when;
+
+            // VirtualSpace
+            JfrNativeEventWriter.putLong(data, 0L); // start
+            JfrNativeEventWriter.putLong(data, 0L); // committedEnd : ulong
+            JfrNativeEventWriter.putLong(data, 0L); // committedSize : ulong
```

Comment:
Just some thoughts: You could try using `GCImpl.getPolicy().getCurrentHeapCapacity()` to record the `committedSize`. I'm not certain, but `reservedSize` [might not be applicable here](https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core.genscavenge/src/com/oracle/svm/core/genscavenge/AbstractCollectionPolicy.java#L375), so maybe it makes the most sense to leave `reservedEnd` and `reservedSize` as 0 for now anyway. You listed these as non-goals so perhaps they can be done at a later time instead.

Reply:
Thank you for the advice. I am not currently able to try it out but I will do so in a few days based on your recommendation.

Reply:
I have added code to output the committed size, and I have verified that the committed size is recorded in jdk.GCHeapSummary.